### DISCUSSION TITLE
Commits merged in PR still appear in a later PR<title>

### DISCUSSION THREAD

**Discussion by huyphung1** [community/community#40891] [Pull Requests] [answered]
### Select Topic Area

Question

### Body

<p>I have a branch name Feature which I commit and create Pull Request to master.<p>
<p>My first PR contains 2 commits A and B that merge to master, on master they appear in a commit that points to that PR.<p>
<p>Then I create a commit C on branch Feature. Now if I create PR or <code>run git log master..Feature</code>, I still got 3 commits in PR to merge to master. I expected to only commit C.<p>
<p>So please explain to me how to do PR correctly because it's really annoying to create a new branch every time. Thanks in advance<p>

**Comment by ossanna16** [comment]
Hi there @huyphung1 and welcome to our community! Thank you for asking a great question 🙂
While you are waiting for a reply to your question, here are a few things you can do:

To get started, introduce yourself in our [official introduction thread](https://github.com/orgs/community/discussions/23204)
To connect with other new users, check out this Discussion to [Meet Your GitHub Community Team](https://github.com/orgs/community/discussions/28776), further your GitHub knowledge by working through our [GitHub Skill 🆙 exercises](https://github.com/orgs/community/discussions/32121), and participate in our [weekly Fun Friday chats](https://github.com/orgs/community/discussions/30912)

**Comment by Preetiraj3697** [comment, marked as answer]
first you check
`git checkout Feature`
`git status`
`git add .`
`git commit -m"add feature"`
`git push origin Feature`
`git checkout master`
`git push origin main`.
Then pull request commit pull request in github.
I hope I explain right.
